#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "motion_controller"


gen = ParameterGenerator()

gen.add("r_start", int_t, 0, "row start", 110, 0, 240)
gen.add("r_end", int_t, 0, "row end", 150, 0, 240)
# r_start < r_end
gen.add("c_start", int_t, 0, "col start", 10, 0, 160)
# c_end = 320 - c_start
gen.add("threshold", int_t, 0, "HoughLines threshold", 50, 30, 100)
gen.add("h_black_up", int_t, 0, "black H lowerbound", 180, 160, 255)
gen.add("v_black_up", int_t, 0, "black V upperbound", 100, 50, 150)
gen.add("motor_status", bool_t, 0, "Motor move or not", False)

exit(gen.generate(PACKAGE, "motion_controller", "corners"))